common/src/main/java/org/keycloak/common/util/MultivaluedHashMap.java

```diff
@@ -101,11 +101,11 @@ public V getFirst(K key)
    {
       List<V> list = get(key);
       if (list == null)
-         put(key, list = new ArrayList<V>());
+         put(key, list = new ArrayList<>());
       return list;
    }
 
-   public void addAll(MultivaluedHashMap<K, V> other)
+   public final void addAll(MultivaluedHashMap<K, V> other)
```

Comment:
Why `final`?

Reply:
Method used in the constructor shouldn't be overridable.

Reply:
Thanks for explanation. I did not look around the PR.